subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/GradleProfilerBuildExperimentRunner.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.performance.fixture;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.gradle.integtests.fixtures.executer.GradleDistribution;
+import org.gradle.internal.UncheckedException;
+import org.gradle.internal.jvm.Jvm;
+import org.gradle.performance.measure.Duration;
+import org.gradle.performance.measure.MeasuredOperation;
+import org.gradle.performance.results.MeasuredOperationList;
+import org.gradle.profiler.BuildAction;
+import org.gradle.profiler.BuildMutatorFactory;
+import org.gradle.profiler.DaemonControl;
+import org.gradle.profiler.GradleBuildConfiguration;
+import org.gradle.profiler.GradleBuildInvoker;
+import org.gradle.profiler.GradleScenarioDefinition;
+import org.gradle.profiler.GradleScenarioInvoker;
+import org.gradle.profiler.InvocationSettings;
+import org.gradle.profiler.Profiler;
+import org.gradle.profiler.RunTasksAction;
+import org.gradle.profiler.instrument.PidInstrumentation;
+import org.gradle.util.GFileUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+// TODO:
+//   - use temporary folder for temp files in Gradle profiler (PidInstrumentation, )
+//   - remove usages of `deleteOnExit` from gradle-profiler
+public class GradleProfilerBuildExperimentRunner implements BuildExperimentRunner {
```

Comment:
it would simplify review if the move and the change would be in separate commits